teamsyncd/teamsync.cpp

```diff
@@ -43,6 +59,32 @@ TeamSync::TeamSync(DBConnector *db, DBConnector *stateDb, Select *select) :
     }
 }
 
+/*
+ * Get interface name based on interface index
+ * Return ifindex in string format if interface don't exist, or no cache
+ *        else return the corresponding interface name.
+ */
+string TeamSync::ifindexToName(int ifindex)
+{
+    char ifName[TeamPortSync::MAX_IFNAME + 1] = {0};
+
+    /* nl_socket, cache create failed earlier */
+    if (!m_nl_sock) {
+        /* Returns ifindex as string / */
+        return to_string(ifindex);
+    }
+ 
+    /* Refill cache and check for ifindex */
+    nl_cache_refill(m_nl_sock ,m_link_cache);
```

Comment:
could return not zero as error indication
Also. Would it be to much to refill all cache for every ifindex to name operation?

Reply:
I will do a bit of profiling and see how much time it takes will all the max number of Port channels created. We need to do a refill before checking -- as I found cache incosistency issues in case when we do config reload command back to back. 

Also I see similar implementations in 
RouteSync::onRouteMsg() --> calling RouteSync::getIfName()
NeighSync::onMsg()  -->  LinkCache::getInstance().ifindexToName()

Reply:
I did more profiling for the refill cache, and found it was done in the same second ( it took around 1ms for refill API to complete execution ) - which looked ok.